lib/chunks.js

```diff
@@ -296,24 +319,46 @@ module.exports.identifyChunksToGenerate = async ({ coursePath, courseId, courseD
         }
 
         await async.each(Object.entries(courseInstanceInfo.assessments), async ([tid, assessmentInfo]) => {
-            const hasClientFilesAssessmentDirectory = await fs.pathExists(
-                path.join(coursePath, 'courseInstances', ciid, 'assessments', tid, 'clientFilesAssessment'),
-            );
+            const assessmentPath = path.join(coursePath, 'courseInstances', ciid, 'assessments', tid);
+            const hasAssessmentDirectory = await fs.pathExists(assessmentPath);
+            const hasClientFilesAssessmentDirectory = await fs.pathExists(path.join(assessmentPath, 'clientFilesAssessment'));
+            if (hasAssessmentDirectory && (
+                (
+                    !existingCourseChunks.courseInstances[ciid] ||
+                    !existingCourseChunks.courseInstances[ciid].assessments ||
+                    !existingCourseChunks.courseInstances[ciid].assessments.has(tid)
```

Comment:
Are we running on Node 14 yet in prod? It'd be so nice to be able to use optional chaining here:
```suggestion
                    !existingCourseChunks.courseInstances[ciid]?.assessments?.has(tid)
```

Reply:
Yes!

```
[ec2-user@ip-10-0-0-162 ~]$ node -v
v14.15.4
```

Reply:
Sweet - just refactored in 25e32a4!